### ISSUE TITLE
anchor link checking is broken

### ISSUE THREAD

**Issue by OmarTawfik** [tcort/markdown-link-check#225]
Thanks for the awesome linter. Long time user, first time contributor.
I have a bunch of files that define HTML code blocks, with `id` attributes:

```html
<code id="foo"> .... </code>
<code id="bar"> .... </code>
````

Then I have some links that reference them:

```html
<a href="#foo"> ... </a>
<a href="../other-file/#bar"> ... </a>
```

After upgrading to the latest version `3.10.3`, it broke CI because of new errors everywhere. I traced it down to the following issues:

1. The linter doesn't collect links from `id` attributes. So any links pointing to it (like `href="#foo"`) produce errors.
2. The linter skips linting `href` attributes that have a file path AND a header.
    1. `href=".#id"` for the same markdown file.
    2. `href="../relative/path#id"` for a markdown file at `../relative/path.md` or `../relative/path/index.md`.

Any feedback/pointers are greatly appreciated. I imagine it is not too hard to fix, given that the linter has partial support for it already.

Although my plate is currently full, I welcome any code pointers/green light on how to fix it, and I can take a stab at it at some point in the future.

**Comment by OmarTawfik** [comment]
cc @tcort any thoughts about how involving the fix is? I imagine it is a simple change, but probably involves a few different packages.

**Comment by MikeMcC399** [comment]
- This is also discussed in issue #195

**Comment by MikeMcC399** [comment]
I can't see any replies from @tcort for the last three months 😵‍💫 so I'm wondering if the repository is still supported. The last commit was 5 months ago.

**Comment by mharding-hpe** [comment]
This is not fixed when checking anchor links in other markdown documents. In those cases, as long as the target document exists, the check passes regardless of whether the target anchor exists. This is causing bad links to get missed in our CI. Would be great if it could be resolved.

**Comment by MikeMcC399** [comment]
@mharding-hpe 

> This is not fixed when checking anchor links in other markdown documents. In those cases, as long as the target document exists, the check passes regardless of whether the target anchor exists. This is causing bad links to get missed in our CI. Would be great if it could be resolved.

Possibly the problem you're seeing is https://github.com/tcort/markdown-link-check/issues/457 where CI reports success even if there are failed links.

**Comment by mharding-hpe** [comment]
@MikeMcC399 No, I'm afraid not (although I have observed that problem as well).

```
mharding@Prepotente:~/linkcheck-test$ cat a.md
[Valid external link](b.md#valid)
[Invalid external link](b.md#invalid)

mharding@Prepotente:~/linkcheck-test$ cat b.md
# Test

## Valid


mharding@Prepotente:~/linkcheck-test$ docker run -v .:/tmp:ro --rm -i ghcr.io/tcort/markdown-link-check:3.13.7 /tmp/a.md

FILE: /tmp/a.md
  [✓] b.md#valid
  [✓] b.md#invalid

  2 links checked.
```

**Comment by MikeMcC399** [comment]
@mharding-hpe 

Thank you very much for the example! That makes it much clearer what you are expecting and it is different from the original post's description. You'd like to have working anchor link checking across local Markdown documents.

It looks to me like it would be a valid enhancement request, so I'd suggest opening a new issue with your example.

**Comment by per1234** [comment]
As @mharding-hpe is already aware, the feature request is already being tracked here: https://github.com/tcort/markdown-link-check/issues/212

---

@OmarTawfik please close this resolved issue. See my previous comment https://github.com/tcort/markdown-link-check/issues/225#issuecomment-2796341013 for definitive proof that the bug has been fixed. It is unfortunate that the maintainers of the project are so rabid about closing issues that they introduced the [incredibly harmful](https://nostalebots.xyz/) "stale bot", but they won't make the effort to close the issues that are actually resolved.

**Comment by MikeMcC399** [comment]
@per1234 

- Thanks for the link to #212! I've hidden my comment as outdated. 
- Hopefully either the original submitter @OmarTawfik will close this issue, or a maintainer, or if nobody posts here again in the next 60 days the stale bot will replace the human inactivity!
